w11b01/test/pgdp/pengusurvivors/GenerateModelTest.java

```diff
@@ -76,6 +76,71 @@ public void fullyConnected3on3() {
         assertEquals(3, maxPairs);
     }
 
+    @Test
+    public void emptyWorkaholics() {
+        int[] workaholics = {};
+        int[] procrastinators = {1, 2, 3};
+        int[][] friendships = {};
+        FlowGraph graph = PenguSurvivors.generateModel(workaholics, procrastinators, friendships);
+        int maxPairs = graph.computeMaxFlowValue();
+        assertEquals(0, maxPairs);
+    }
+
+    @Test
+    public void emptyProcrastinators() {
+        int[] workaholics = {1, 2, 3};
+        int[] procrastinators = {};
+        int[][] friendships = {};
+        FlowGraph graph = PenguSurvivors.generateModel(workaholics, procrastinators, friendships);
+        int maxPairs = graph.computeMaxFlowValue();
+        assertEquals(0, maxPairs);
+    }
+
+    @Test
+    public void nullWorkaholics() {
+        int[] workaholics = null;
```

Comment:
null cases aren't specified in the task and are not tested by the Artemis tests. I would recommend to leave them out for sake of simplicity.

Reply:
👍